rockcraft/extensions/springboot.py

```diff
@@ -0,0 +1,291 @@
+                logpath_report=False,
+            )
+
+        if (
+            self.user_gradle_init_script_part_override_build_override
+            and not self.build_gradle_path.exists()
+        ):
+            raise ExtensionError(
+                "gradle init script part is enabled for non-gradle build",
+                doc_slug="/reference/extensions/spring-boot-framework",
+                logpath_report=False,
+            )
+
+    @property
+    def mvnw_path(self) -> pathlib.Path:
+        """Return the path to the mvnw file."""
+        return self.project_root / "mvnw"
+
+    @property
+    def pom_xml_path(self) -> pathlib.Path:
+        """Return the path to the pom.xml file."""
+        return self.project_root / "pom.xml"
+
+    @property
+    def gradlew_path(self) -> pathlib.Path:
+        """Return the path to the gradlew file."""
+        return self.project_root / "gradlew"
+
+    @property
+    def build_gradle_path(self) -> pathlib.Path:
+        """Return the path to the build.gradle file."""
+        return self.project_root / "build.gradle"
+
+    @property
+    def _rock_base(self) -> str:
+        """Return the base of the rockcraft project."""
+        return self.yaml_data["base"]
+
+    @property
+    def user_gradle_init_script_part_override_build_override(self) -> str:
+        """Return the user's override-build part for gradle-init-script part."""
+        return (
+            self.yaml_data.get("parts", {})
+            .get("spring-boot-framework/gradle-init-script", {})
+            .get("override-build", "")
+        )
+
+    def gen_gradle_init_script_part(self) -> dict[str, Any]:
+        """Generate the gradle-init-script part if required."""
+        if not self.user_gradle_init_script_part_override_build_override:
+            return {}
+        return {
+            "spring-boot-framework/gradle-init-script": {
+                "plugin": "nil",
+                "source": ".",
+                "override-build": self.user_gradle_init_script_part_override_build_override,
+            }
+        }
+
+    def gen_install_app_part(self) -> dict[str, Any]:
+        """Generate the install-app part."""
+        if self.pom_xml_path.exists() and self.build_gradle_path.exists():
+            raise ExtensionError(
+                "Both mvnw and gradlew exist. Please remove one of them."
+            )
+
+        plugin: Literal["gradle", "maven"] = (
+            "gradle" if self.build_gradle_path.exists() else "maven"
+        )
+        install_app_part = {
+            "source": ".",
+            "organize": {"**/*.jar": "app/"},
+            **(
+                self._gen_install_app_gradle_plugin()
+                if plugin == "gradle"
+                else self._gen_install_app_maven_plugin()
+            ),
+        }
+
+        if self._rock_base == "bare":
+            install_app_part["stage-packages"] = ["zlib1g", "libstdc++6"]
```

Comment:
I assume those are installed as JDK dependencies, so we can borrow the list of slices from https://github.com/canonical/chisel-releases/blob/ubuntu-24.04/slices/openjdk-21-jre-headless.yaml 

```suggestion
            install_app_part["stage-packages"] = ["libnss3_nss", "libc6_libs", "libgcc-s1_libs", "libstdc++6_libs", "zlib1g_libs" ]
```